packages/firestore/src/local/remote_document_change_buffer.ts

```diff
@@ -66,23 +75,16 @@ export abstract class RemoteDocumentChangeBuffer {
     transaction: PersistenceTransaction
   ): PersistencePromise<void>;
 
-  protected set readTime(value: SnapshotVersion) {
-    // Right now (for simplicity) we just track a single readTime for all the
-    // added entries since we expect them to all be the same, but we could
-    // rework to store per-entry readTimes if necessary.
-    debugAssert(
-      this._readTime === undefined || this._readTime.isEqual(value),
-      'All changes in a RemoteDocumentChangeBuffer must have the same read time'
-    );
-    this._readTime = value;
-  }
-
-  protected get readTime(): SnapshotVersion {
-    debugAssert(
-      this._readTime !== undefined,
-      'Read time is not set. All removeEntry() calls must include a readTime if `trackRemovals` is used.'
-    );
-    return this._readTime;
+  protected getReadTime(key: DocumentKey): SnapshotVersion {
```

Comment:
Is there a reason you changed this from a property to a method?

Reply:
Because it takes a parameter now?